build failed on running flow unit test with JDK 21 ea

### Description of the bug

after #17240 , when running `mvn test` on flow project, it failed in many classes with following errors on `Flow HTML Components`

```
 IllegalState Property information for 'className' missing

[ERROR] Tests run: 426, Failures: 0, Errors: 422, Skipped: 0
```



### Expected behavior

build passes

### Minimal reproducible example

- take the current `main` branch ([9364ed2](https://github.com/vaadin/flow/commit/9364ed2b25a95e8fc250d4b54181758dc8ad3968))
- run ` mvn -e -V -T 8 test -am -pl flow-html-components,flow-dnd,flow-plugins -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn`

### Versions

- Vaadin / Flow version: 24.2-SNAPSHOT
- Java version: \openjdk-21-ea+31
- OS version: WINDOWS

--- a/flow-html-components/src/test/java/com/vaadin/flow/component/html/ComponentTest.java
+++ b/flow-html-components/src/test/java/com/vaadin/flow/component/html/ComponentTest.java
@@ -408,6 +408,12 @@ public abstract class ComponentTest {
     private void assertProperty(PropertyDescriptor descriptor) {
         if (descriptor.getReadMethod() != null
                 && descriptor.getWriteMethod() != null) {
+            // JDK 21 detects properties also from interface default methods
+            // Test setup can safely ignore properties from mixin interfaces
+            if (descriptor.getReadMethod().isDefault()
+                    && descriptor.getWriteMethod().isDefault()) {
+                return;
+            }
             if (!hasProperty(descriptor.getName())) {
                 throw new IllegalStateException("Property information for '"
                         + descriptor.getName() + "' missing");

